From aba5a8a85249f88303a86cea33a64631a0a97cdc Mon Sep 17 00:00:00 2001
From: Harninder Rai <harninder.rai@freescale.com>
Date: Thu, 30 May 2013 20:53:01 +0530
Subject: [PATCH 370/547] eSDHC: Add Quirks for Multi block read and timeout

On BSC9132QDS platform, continuous error messages come on linux
because of errata related to multi block read and ACMD timeout
This patch adds SDHCI_QUIRK_BROKEN_TIMEOUT_VAL and
SDHCI_QUIRK_MULTIBLOCK_READ_ACMD12 quirks to fix this issue

Signed-off-by: Harninder Rai <harninder.rai@freescale.com>
Change-Id: Ie59a954af2c895f86c58a16cdc8865f8f558c5bb
Reviewed-on: http://git.am.freescale.net:8181/2793
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/mmc/host/sdhci-of-esdhc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-of-esdhc.c b/drivers/mmc/host/sdhci-of-esdhc.c
index b0dc026..9493920 100644
--- a/drivers/mmc/host/sdhci-of-esdhc.c
+++ b/drivers/mmc/host/sdhci-of-esdhc.c
@@ -417,7 +417,9 @@ static const struct sdhci_pltfm_data sdhci_esdhc_pdata = {
 	 */
 	.quirks = ESDHC_DEFAULT_QUIRKS | SDHCI_QUIRK_BROKEN_CARD_DETECTION
 		| SDHCI_QUIRK_NO_CARD_NO_RESET
-		| SDHCI_QUIRK_NO_ENDATTR_IN_NOPDESC,
+		| SDHCI_QUIRK_NO_ENDATTR_IN_NOPDESC
+		| SDHCI_QUIRK_BROKEN_TIMEOUT_VAL
+		| SDHCI_QUIRK_MULTIBLOCK_READ_ACMD12,
 	.ops = &sdhci_esdhc_ops,
 };
 
-- 
1.8.4.93.g57e4c17

